File: unit-tests/switch.c
typedef enum { A, B, C } e;

int _main() {
  int r = 0;
  switch (A) {
    case B:
    case C:
      r = 1;
      break;
    case A:
      r = 0;
      break;
  }
  return r;
}

File: unit-tests/switch.rs
#![allow(non_snake_case)]
#![allow(non_upper_case_globals)]
#![allow(non_camel_case_types)]
#![allow(unused_assignments)]
#![allow(unreachable_patterns)]
#![allow(unused_mut)]

pub fn _main() -> i32
{
  let mut r: i32 = 0i32;
  match e::A
  {
    e::B => r = 1i32,
    e::C => r = 1i32,
    e::A => r = 0i32,
    _ => panic!("Incomplete pattern matching")
  };
  return r
}

#[derive(PartialEq, Clone, Copy)]
pub enum e
{
  A,
  B,
  C
}
fn main() { assert_eq!(0, _main()) }
